### DISCUSSION TITLE
Using a Computed Properties to simplify relation

### DISCUSSION THREAD

**Discussion by rodolphonetto** [adonisjs/core#1279] [Help]
Hi guys, I've a relation where in the father I've the id and the child (details) I've the title. They are related using a hasOne, so I need to put a .with('details') to see the title, but I don't want to enter the details.title to get the title, there is a way to set a computed propertie in the model where it send the title directly?

**Comment by fnoquiq** [comment]
I didn't understand what you're trying to do. Which version are you using?

**Reply by rodolphonetto** [reply]
Hey, I'll try to explain it better

`                "selected_answers": [
                    {
                        "id": 16,
                        "form_id": 2,
                        "answer": null,
                        "answer_id": 1,
                        "additional_answer": null,
                        "details": {
                            "id": 1,
                            "form_id": 2,
                            "title": "Opção 1",
                        }
                    }
                ]`

selected_answers is a model. I want to set a computed to show the details.title in the root like that

`                "selected_answers": [
                    {
                        "id": 16,
                        "form_id": 2,
                        "answer": null,
                        "answer_id": 1,
                        "additional_answer": null,
                        "title": "Opção 1",
                    }
                ]`

Is it clear now? thank for you time :)

**Reply by fnoquiq** [reply]
I believe you can do this with hooks, but it will result in a loss of performance. 

Hooks can be async while computed properties aren't. You can find more about hooks [here](https://preview.adonisjs.com/guides/models/hooks).

```ts
import { hasOne, HasOne, computed, afterFind } from '@ioc:Adonis/Lucid/Orm'

@hasOne(() => Details)
public details: HasOne<typeof Details>

@computed()
public get title() {
  return this.details.title
}

@afterFind()
public static async afterFindHook(selectedAnswers: SelectedAnswers) {
  await selectedAnswers.preload('details')
}
```

Note: I can't tell you if this is the best way to do this

**Comment by McSneaky** [comment]
Hmm, I don't think there's something like that, but you can merge two objects together
```js
const newObject = {
  ...objectOne,
  ...objectTwo // careful, if keys collide with object one, then it will overwrite those
}
```
Or use [Object.assign()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)

**Reply by rodolphonetto** [reply]
I tried to explain it again in the answer above, maybe it is more clear on what I'm trying to do

**Comment by iammrsea** [comment]
I think there could be a workaround.
You could add a title computed property in your selected_answers model as follows:

```javascript

@hasOne(()=>Details,{serializedAs: null))
public details: HasOne<typeof Details>

@computed()
public get title(){
 if(this.details){
 return this.details.title
}return null
```
Note, the **serializeAs: null** makes sure that details field is not returned with the selected_answer model

**Reply by rodolphonetto** [reply]
I'll try it, I'm using adonis 4.1 this would work there as well?

**Reply by fnoquiq** [reply]
No, the example is in the latest version (v5).

**Reply by rodolphonetto** [reply]
Before everything I wanna say thanks for the time you all are spending with me, thanks :)

I tried to do with v4.1 but it does not work what do u think I can do?


```
 static get computed() {
    return ['slug']
  }

  async getSlug({ format_id }) {
    const format = await FormFormat.findOrFail(format_id)
    console.log(format.slug)
    return format.slug
  }
```


the console.log is return the slug I need but the return on postman is like "slug": {},

**Reply by fnoquiq** [reply]
Can you put the controller code?

---

Try to call `.toJSON()`
```ts
const json = user.toJSON()
console.log(json.slug)
```
